stage #4/module #2. REST API Advanced/rest_api_advanced.md

```diff
@@ -18,105 +18,124 @@
 The recommended timeline for the whole module is 2 weeks.
 
 ### Task
+
+#### Part 1
+
+##### Business requirements
+
+1. Develop web service for News Management system with the following entities:
+   ![Schema](./media/schema.jpg)
+   * <img src="./media/snowflake.jpg" alt="Snowflake" width="18"/> - unique value
+   * All Name, Title and Content fields are required
+   * Created, Modified – have ISO 8601 format [wiki: ISO_8601](https://en.wikipedia.org/wiki/ISO_8601). Example: 2018-08-29T06:12:15.156. More discussion here: [stackoverflow: how to get iso 8601](https://stackoverflow.com/questions/3914404/how-to-get-current-moment-in-iso-8601-format-with-date-hour-and-minute).
+     
+2. The system should expose REST APIs to perform the following operations:
+   * CUD operations ([what is CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete)) for News and Comment. If new tags and authors are passed during creation/modification – they should be created in the DB. For update operation – update only fields, that pass in request, others should not be updated. Batch insert is out of scope.
+   * Get News:
+     * get all
+     * by Id
+     * search (all params are optional and can be used in conjunction):
+       * by tag names and tag ids (many tags)
+       * by author name (one author)
+       * by part of Title
+       * by part of Content
+     * sort by Created, Modified Asc/Desc. Default: Created Desc
+   * Get Comments:
+     * by News Id (URL example: /news/{newsId}/comments)
+     * by Id
+     * sort by Created, Modified Asc/Desc. Default: Created Desc
+   * CRD operations for Tag, and Author
+   * Get Tags and Authors:
+     * get all
+     * by Id
+     * by part Name
+     * by News Id (URL examples:\
+           /news/{newsId}/tags - *should return tags collection*\
+           /news/{newsId}/authors - *should return 1 author*)
```

Comment:
if return 1 author, that need /news/{newsId}/author

Reply:
updated